### ISSUE TITLE
[miele] reading state at startup and reading state change not working

### ISSUE THREAD

**Issue by Alex5719** [openhab/openhab-addons#6311]
In snapshot build 2.5.0~S1736-1 the state of the hood light is not read at startup (is still NULL).

**Comment by Alex5719** [comment]
I downgraded my openHAB installation to the 2.5.0 M4 build and can confirm that in this older version the miele binding is reading the state of the hood light at startup correctly.

**Comment by Alex5719** [comment]
Can someone look into this issue please? It seems that the latest commits on the miele binding have broken the functionality at least for Miele Zigbee devices.

**Comment by Alex5719** [comment]
Not only reading the state at startup is affected. When manually changing the hood light directly at the device the state changes are also not read.

**Comment by Alex5719** [comment]
The new milestone build 2.5.0 M5 is also affected by the bug. The state of the light of the hood is not read on startup of openhab and also not read when turning the light on and off manually by using the HW button at the hood. Switching the hood light on and off via the openhab switch is working. The Miele binding is not throwing any exceptions. Reading the state at startub and on state change did work in the old milestone build 2.5.0 M4.

**Comment by kaikreuzer** [comment]
> The new milestone build 2.5.0 M5 is also affected by the bug. 

Yes, clearly, as it is in the snapshots as you have reported and nobody looked into it / fixed it.

@MartinLepsy Can you confirm the issue? If so, are you able to fix it?

**Comment by MartinLepsy** [comment]
@kaikreuzer I'll try to have a look on the issue during this week. If I am able to reproduce it i will try to fix it, too.

**Comment by MartinLepsy** [comment]
Hello @kaikreuzer @Alex5719 - unfortunately I was unable to reproduce this behaviour.
I tested it with my development environment and additionally with openhabian (2.5.0~S1759-1) together with a ZigBee hood and additionally a ZigBee washing machine. 

The status was always displayed correctly and changed according to my actions. There were no problems with reading the state information at startup.

What hood do you have? Can you provide any more information?

**Comment by Alex5719** [comment]
Hello MartinLepsy,

thank you fpr looking into my issue. The hood is indeed a special model: DA 6396 W Puristic ECO. The gateway has the software version 2.4.0 installed and is showing the following device details of the hood:

Technical Type | not defined
Electronic Software ID | 0
Fabrication Number | not defined
Interface | XKS 3000 Z

Actually the hood is not my only Miele device and the other devices are also not read. I now unregistered all devices from the gateway and only registered the oven. 
And still the state of my Miele oven is not read by the new binding.

Nevertheless when I switch back to the older version of the binding everything works.

**Comment by Alex5719** [comment]
I am using config files for my things and items

**Comment by MartinLepsy** [comment]
Any chance to get a snippet or the whole config file? So far I always added all things via paper UI so it may be worth analyzing.

**Comment by Alex5719** [comment]
I now commented out the miele bridge and miele things in my thing config file. The used config is really standard almost identical to the example in the binding documentation. The only difference is, that the names of the things are chosen to be identical to the names used by the auto discovery function of paper ui. I also removed the miele gateway and things via paper ui. After adding the miele gateway and the miele things using the auto discovery function of paper ui the new binding worked correctly. After again adding the miele bridge and things to the thing config file the binding still works now.

It seems that the new version of the binding is now working for me. I am unsure what other users needs do to migrate from the old version of the miele binding to the new version
